### PR TITLE
Extension storage

### LINKED ISSUE #35: Local storage / Cache - plugin/user information

### PR DESCRIPTION
**Pull request by jolampi**
Fixes #35

### CODE REVIEW

File: src/config/storage.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { ExtensionContext } from "vscode";
```
Comment (jesseanttila-cai) [inline src/config/storage.ts]: Consider using `import * as vscode from "vscode"` for uniformity

File: src/test/config/storage.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import * as assert from "assert";
+import * as TypeMoq from "typemoq";
+import * as vscode from "vscode";
+
+import Storage from "../../config/storage";
+
+suite("Storage tests", () => {
+
+    const createMocks = () => {
```
Comment (jesseanttila-cai) [inline src/test/config/storage.test.ts]: Arrow syntax used for a named function here, I believe they were to be reserved for anonymous functions for now

Comment (jolampi) [reply src/test/config/storage.test.ts]: Fair enough. This was the style used in default tests so I followed it in the absence of existing examples.

File: src/config/storage.ts (hunk 1 above)
Comment (jolampi) [reply src/config/storage.ts]: In that case this should be defined as a convention going forward.

File: src/test/config/storage.test.ts (hunk 2 above)
Comment (jolampi) [reply src/test/config/storage.test.ts]: The current setup of TSLint complains if these arrows are turned into function syntax.

**Pull request merged**